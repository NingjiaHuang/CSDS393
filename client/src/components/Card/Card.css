.card-container{
    width: 300px;
    overflow: hidden;
    box-shadow: 0px 0px 15px -5px;
    transition: 0.5s;
    animation: ease-in;
    float: left;
    margin-left: 2rem;
    margin-top: 2rem;
}

.card-container:hover{
    transform: scale(1.1);
    box-shadow: 0px 0px 15px 0px;
}

.image-container img{
    overflow: hidden;
    height: 200px;
}

.card-content{
    margin: 1rem;
    margin-top: 0.3rem;
}

h3, p{
    margin: 0;
    padding: 0;
}

.card-title{
    margin-bottom: 0.5rem;
}

.btn{
    display: flex;
    justify-content: space-between;
}

.btn button{
    padding: 0.5rem;
    background-color: white;
    border: none;
    transition: 0.2s;
    margin:0.5rem;
    margin-bottom: 0.5rem;
    border-radius: 3px;
    transform: scale(1.0);
}

.btn button:hover{
    background: rgb(191, 145, 147, 0.1)
}

a{
    text-transform: uppercase;
    color: rgb(191, 145, 147);
    text-decoration: none;
    font-weight: bold;
    position: center;
}
