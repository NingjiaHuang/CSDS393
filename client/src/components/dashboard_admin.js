import React, { Fragment, useState, useEffect } from "react";
import Card from "./Card/Card"
import manage from "../img/manageaccount.png"

const AdminDashboard = ({setAuth}) => {

    const [name, setName] = useState("")
    async function getName() {
        try{
            const response = await fetch("http://localhost:4020/admin_dashboard/", {
                method: "GET",
                headers:{token: localStorage.token}
            });
            const parseRes = await response.json()
            setName(parseRes.username)
        }catch(err){
            console.error(err.message);
        }
    }

    const logout = (e) => {
        e.preventDefault();
        localStorage.removeItem("token");
        setAuth(false);
    }

    useEffect(() => {
        getName()
    })
    return (
        <Fragment>
            <h1>Administrator Dashboard {name}</h1>
            <button data-testid ="logoutbtn" className="btn btn-primary" onClick={e => logout(e)}>Logout</button>
            <Card title='Manage Account' imageUrl={manage} body='Manage existing breeder and parent accounts.' button="Manage Account" url="/manage_user_account"/>
        </Fragment>
    )
}

export default AdminDashboard;