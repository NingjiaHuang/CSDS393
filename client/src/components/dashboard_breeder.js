import React, { Fragment, useState, useEffect } from "react";
import Card from "./Card/Card"
import add from "../img/add.png"
import edit from "../img/edit.png"
import gene from "../img/gene.png"
import tree from "../img/tree.png"

const BreederDashboard = ({setAuth}) => {

    const [name, setName] = useState("")
    async function getName() {
        try{
            const response = await fetch("http://localhost:4020/breeder_dashboard/", {
                method: "GET",
                headers:{token: localStorage.token}
            });
            const parseRes = await response.json()
            setName(parseRes.username)
        }catch(err){
            console.error(err.message);
        }
    }

    const logout = (e) => {
        e.preventDefault();
        localStorage.removeItem("token");
        setAuth(false);
    }

    useEffect(() => {
        getName()
    })
    return (
        <Fragment>
            <div>
                <h1>Breeder Dashboard {name}</h1> 
                <button data-testid="logoutBtn" className="btn btn-primary" onClick={e => logout(e)}>Logout</button>
                <Card title='Add Cat Info' imageUrl={add} body='Add new cats to the existing database.' button="Add Cat"/>
                <Card title='Edit Cat Info' imageUrl={edit} body='Edit existing cat information.' button="Edit Cat"/>
                <Card title='Gene Calculator' imageUrl={gene} body='Get info of potential genes of kitten.' button="Gene Calculator"/>
                <Card title='Family Tree' imageUrl={tree} body='Check the ancestors and descendents of cat.' button="Family Tree"/>
            </div>
        </Fragment>
    )
}

export default BreederDashboard;