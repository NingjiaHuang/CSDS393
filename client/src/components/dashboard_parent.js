import React, { Fragment, useState, useEffect } from "react";
import Card from "./Card/Card"
import cattery from "../img/add.png"
import edit from "../img/edit.png"
import gene from "../img/gene.png"
import tree from "../img/tree.png"
const ParentDashboard = ({setAuth}) => {
    const [name, setName] = useState("")
    async function getName() {
        try{
            const response = await fetch("http://localhost:4020/parent_dashboard/", {
                method: "GET",
                headers:{token: localStorage.token}
            });
            const parseRes = await response.json()
            setName(parseRes.username)
        }catch(err){
            console.error(err.message);
        }
    }

    const logout = (e) => {
        e.preventDefault();
        localStorage.removeItem("token");
        setAuth(false);
    }

    useEffect(() => {
        getName()
    })
    return (
        <Fragment>
            <h1>Parent Dashboard, {name}</h1>
            <button className="btn btn-primary" onClick={e => logout(e)}>Logout</button>
            <Card title='View Cat Info' imageUrl={edit} body='View existing cat information in the database.' button="Edit Cat" url="/manage_cat_parent"/>
            <Card title='Gene Calculator' imageUrl={gene} body='Get info of potential genes of kitten.' button="Gene Calculator" url="/gene_calc"/>
            <Card title='Family Tree' imageUrl={tree} body='Check the ancestors and descendents of cat.' button="Family Tree" url="/family_tree"/>
            <Card title='Search Cattery' imageUrl={cattery} body='Search the catteries who use our system.' button="Search Cattery" url="/search_cattery"/>
        </Fragment>
    )
}

export default ParentDashboard;