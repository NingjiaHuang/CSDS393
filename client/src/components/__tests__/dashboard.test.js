import {render, screen, fireEvent, cleanup} from '@testing-library/react';
import Card from '../Card/Card';
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect';
import dashboard_admin from '../dashboard_admin'
import dashboard_breeder from '../dashboard_breeder'
import dashboard_parent from '../dashboard_parent'
afterEach(() =>{
    cleanup();
})

test('matches snapshot of a card', () =>{
    const tree = renderer.create(<Card/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('matches snapshot of parent dashboard', () =>{
    const tree = renderer.create(<dashboard_parent/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('matches snapshot of breeder dashboard', () =>{
    const tree = renderer.create(<dashboard_breeder/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('matches snapshot of admin dashboard', () =>{
    const tree = renderer.create(<dashboard_admin/>).toJSON();
    expect(tree).toMatchSnapshot();
})