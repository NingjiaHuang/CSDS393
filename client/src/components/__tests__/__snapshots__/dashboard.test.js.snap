// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of a card 1`] = `<card />`;

exports[`matches snapshot of admin dashboard 1`] = `<dashboard_admin />`;

exports[`matches snapshot of breeder dashboard 1`] = `<dashboard_breeder />`;

exports[`matches snapshot of parent dashboard 1`] = `<dashboard_parent />`;
